src/convertToObject.js

```diff
@@ -1,18 +1,23 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedStyles = sourceString
+    .split(';')
+    .map(item => item.trim())
+    .filter(item => item !== '')
+    .map(item => item.split(':')
+      .map(value => value.trim()))
+    .reduce((prev, curr) => ({
+      ...prev,
```

Comment:
Wow, 5 iterations. Not the best performance, let's optimize this algorithm and made it with only 1 iteration.

Reply:
I was only able to remove one map method. I have no idea how I can do this in one iteration

Reply:
@illiashcherbak imagine if you will have such a task, to optimize the productivity of some solution, what will you do? 
Saying "I have no idea" is not the best solution. Why didn't you come to Q&A if you don't know how to do it?

I will wait for you on my Q&A tomorrow at 13:30 and we will discuss how to resolve it in 1 iteration.